compiler/semtypes.nim

```diff
@@ -1261,6 +1261,14 @@ proc isMagic(sym: PSym): bool =
   let nPragmas = sym.ast[pragmasPos]
   return hasPragma(nPragmas, wMagic)
 
+proc maybeForkType(c: PContext, t: PType): PType =
+  if t.kind in {tyOr} and t.sym != nil:
```

Comment:
Seems to me you're looking for:

```nim

if t.kind == tyOr: result = copyType(t)
else: result = t

```

Reply:
this solution does't work. the type used in type equality and symbol binding. one of tyor's son get isNone, so here the symbol binding need extro information to distinguish

Reply:
use temporary solution mitigate the problem